plugins/inputs/windows_event_log/wineventlog/wineventlog_test.go

```diff
@@ -171,6 +326,69 @@ func checkEvents(t *testing.T, records []*windowsEventLogRecord, substring strin
 	assert.Equal(t, count, found, "expected %v, %v, actual %v", substring, count, found)
 }
 
+func marshalRangeList(rl state.RangeList) string {
+	var marshalledRanges []string
+	for _, r := range rl {
+		text, _ := r.MarshalText()
+		marshalledRanges = append(marshalledRanges, string(text))
+	}
+	return strings.Join(marshalledRanges, ",")
+}
+
+func assertStateFileRange(t *testing.T, fileName string, rl state.RangeList) {
+	content, _ := os.ReadFile(fileName)
+	assert.Contains(t, string(content), marshalRangeList(rl))
+}
```

Comment:
nit: Instead of marshaling, we could also just call the `Restore` function of the state manager.